### PR TITLE
Feat/model lazy loading

### PR DESCRIPTION
**Pull request by AleksMat**
Changes:
- Lazy loading of LightGBM model
- LightGBM model is now being saved in native format and only `lightgbm.Booster` is used for prediction

### CODE REVIEW

File: s2cloudless/PixelClassifier.py [hunk 1]
```diff
@@ -88,9 +88,12 @@ def image_predict_proba(self, X):
...
         """
-
         pixels = self.extract_pixels(X)
 
-        probabilities = self.classifier.predict_proba(pixels)
+        if hasattr(self.classifier, 'predict_proba'):
+            probabilities = self.classifier.predict_proba(pixels)
+        else:
+            probabilities = self.classifier.predict(pixels)
+            probabilities = np.vstack((1. - probabilities, probabilities)).transpose()
```
Comment (mlubej) [inline s2cloudless/PixelClassifier.py]: I don't understand this. Which classifier does this?

Comment (AleksMat) [reply s2cloudless/PixelClassifier.py]: In previous versions, when we used pickled model, we had here an instance of `lightgbm.LGBMClassifier` class. It had methods `predict` and `predict_proba`. Now the model is loaded in `lightgbm.Booster` class and it has only method `predict`, which works differently than `LGBMClassifier.predict`.

I have just got an idea of how to write this in a much more explicit way -> will change.

Comment (AleksMat) [reply s2cloudless/PixelClassifier.py]: Change added

File: s2cloudless/PixelClassifier.py [hunk 2]
```diff
@@ -57,23 +62,26 @@ def extract_pixels(X):
...
 
         :return: raster classification map
         :rtype: numpy array, [n_samples, n_pixels_y, n_pixels_x]
         """
-
         pixels = self.extract_pixels(X)
 
+        if isinstance(self.classifier, Booster):
+            raise NotImplementedError('An instance of lightgbm.Booster can only return prediction probabilities, '
+                                      'use PixelClassifier.image_predict_proba instead')
```
Comment (mlubej) [inline s2cloudless/PixelClassifier.py]: probably typo? `image_predict_proba` -> `image_predict`

**Pull request merged**